faiss/gpu/impl/IVFBase.cu

```diff
@@ -677,9 +680,209 @@ int IVFBase::addVectors(
             listOffsetDevice,
             stream);
 
+    // save index mapping
+    HostTensor<Index::idx_t, 1, true> indicesHost(indices, stream);
+    indexMapping(listIdsHost, listOffsetHost, indicesHost);
+
     // We added this number
     return numAdded;
 }
 
+void IVFBase::removeVectors_(
+        Tensor<int, 1, true>& listIds,
+        Tensor<int, 1, true>& listOffset,
+        Tensor<int, 1, true>& listReplaceOffset,
+        Tensor<long, 1, true>& listIndicesReplaceOffset,
+        cudaStream_t stream) {
+    FAISS_ASSERT_MSG(false, "need to be implemented in subclasses");
+}
+
+size_t IVFBase::removeVectors(size_t n, const Index::idx_t* indices) {
```

Comment:
can you indicate in the name that `indices` is a host pointer (e.g., `indicesHost`)?

Also, looks like the case INDICES_CPU is not handled, can this case be dealt with as well?